src/GeniusYield/Test/FeeTracker.hs

```diff
@@ -124,7 +135,20 @@ wrapBodyBuilder f skeletons = do
                     in ownLostDeposit <> otherGainedDeposit
                 )
                 outsWithBlueprint
-        modify' (\prev -> prev <> feeExtraLovelace <> depositsExtraLovelace)
+        modify' (\prev -> prev <> feeExtraLovelace <> depositsExtraLovelace <> unexpectedGainExtraLovelace)
+    unexpectedGainOf pp utxo
+      -- UTxO contains exactly the minimum amount of Ada.
+      -- We assume this was added by the balancer to cover min ada and not expected as an effective "gain" by the user.
+      -- So we track it along.
+      -- TODO (chase): It is possible that the user expected a portion of this Ada.
+      -- (i.e the output was originally created with 1 Ada but the real output contains 2 Ada to cover min ada)
+      -- This case is ambiguous. So there should be a hint mechanism in place to try and help the fee tracker with this.
```

Comment:
I envision this mechanism would be quite useful. But I didn't have time to design and implement it yet.

Reply:
Isn't it straight forward to fix? We already have tx skeletons where we know ada expected by user.

Reply:
We don't know how much ada a user is expecting from an input. Only outputs.

We already use the output information on an individual basis. We _could_ remember the outputs created by users when they are being used as an input later on. So we know what lovelace is expected from that input according to previous transactions skeleton.

This doesn't cover all cases but it likely covers a vast majority. This is actually supposed to be part of the hint mechanism (automatically add an hint)

Reply:
I understand, let's move forward with this change and follow up on improvint it.